radixdlt-java/src/main/java/com/radixdlt/client/application/translate/FeeProcessor.java

```diff
@@ -33,9 +36,14 @@
 	 * Processes actions in the context of fee generation.
 	 *
 	 * @param actionProcessor An {@link ActionProcessor} to use to process any actions associated with the fee
-	 * @param metadataProcessor A {@link MetadataProcessor} to use to process any metadata associated with the fee
 	 * @param address The address of the fee payee
 	 * @param feelessAtom The atom, without fees, for which to generate fees
+	 * @param optionalFee The fee proposed by the client, if present
 	 */
-	void process(ActionProcessor actionProcessor, MetadataProcessor metadataProcessor, RadixAddress address, Atom feelessAtom);
+	void process(
+		ActionProcessor actionProcessor,
+		RadixAddress address,
+		Atom feelessAtom,
+		Optional<BigDecimal> optionalFee
```

Comment:
why not UInt256?

Reply:
It could be, but that would just require converting to subunits and then back again, as the burn API is in units, and we would like all our user-facing APIs to be in units as well.

For clarity, the BigDecimal (unit) values are the UInt256 (subunit) values multiplied by 10^-18.  See TokenUnitConversions for more details.